### ISSUE TITLE
Fatal error: Call to undefined method Joomla\Ldap\LdapClient::escape()

### ISSUE THREAD

**Issue by FEAGRI** [joomla/joomla-cms#18072]
### Steps to reproduce the issue
Log in to the site / front-end with LDAP user or Joomla-registered user
Log in by admin / backend with LDAP user or Joomla registered user

### Expected result
Access to the intranet / front-end website or access to the joomla administrative area or back-end

### Actual result
Fatal error: Call to undefined method Joomla\Ldap\LdapClient::escape() in /var/www/html/portal/plugins/authentication/ldap/ldap.php on line 90

### System information (as much as possible)
I noticed this difference without code from the ldap.php file from version 3.7.5 to 3.8.0 and I did not understand the error:

j!3.7.5, lines:
84				if ($bindtest)
85				{
86					// Search for users DN
87					$binddata = $ldap->simple_search(str_replace('[search]', $credentials['username'], $this->params->get('search_string')));
88
89					if (isset($binddata[0], $binddata[0]['dn']))

j!3.8.0, lines:
84				if ($bindtest)
85				{
86					// Search for users DN
87					$binddata = $ldap->simple_search(
88						str_replace(
89							'[search]',
90							$ldap->escape($credentials['username'], **null**, **LDAP_ESCAPE_FILTER**),
91							$this->params->get('search_string')
92						)
93					);
94
95					if (isset($binddata[0], $binddata[0]['dn']))

### Additional comments
Basically, more parameters were added to $ binddata = $ ldap-> simple_search (
on line 87, the same happens for line 119 and 123 in j! 3.8.0
**obs**: Our portal is based on LDAP connection, all users use this authentication model, only super admin users who do not!
**And even the super admin does not connect**

Tanks!

**Comment by alikon** [comment]
can help #17989

**Comment by ghost** [comment]
closed as duplicated Report #17989<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18072">issues.joomla.org/tracker/joomla-cms/18072</a>.</sub>
